includes/class-newspack-newsletters-subscription.php

```diff
@@ -1013,7 +1047,11 @@ public static function process_subscription_update() {
 		} else {
 			$email  = get_userdata( get_current_user_id() )->user_email;
 			$lists  = isset( $_POST['lists'] ) ? array_map( 'sanitize_text_field', $_POST['lists'] ) : [];
-			$result = Newspack_Newsletters_Contacts::update_lists( $email, $lists, 'User updated their subscriptions on My Account page' );
+			if ( self::is_newsletter_subscriber( $email ) ) {
```

Comment:
tid: Here we are relying that `WP_Error` is evaluated as `truthy`. Nothing wrong with that, but from a readability perspective it's a little weird and fragile.

If we inverted the order and id `if ( false === self::is_newsletter_subscriber( $email ) ) {` it would look much more clear. WDYT?

Reply:
Done in 6c98249949d88702967b2c3417ebf5bace728d42